File: Sem2/PO_projekt/Java/plants/NightShade.java
package Java.plants;

import Java.Game;
import Java.Organism;
import Java.Plant;
import java.awt.Color;

public class NightShade extends Plant {

    public NightShade(int x, int y, Game game) {
        super(99, 0, x, y, game);
    }

    @Override
    public Color getColor() {
        return new Color(128, 0, 128); // purple
    }

    @Override
    public Organism getOrganism() {
        return this;
    }

    @Override
    protected Plant spread(int targetX, int targetY) {
        return new NightShade(targetX, targetY, game);
    }

    @Override
    protected void collision(Organism attacker) {
        game.setCell(attacker.getX(), attacker.getY(), null);
        game.organisms.remove(attacker);
        
        super.collision(attacker);
    }
}

File: Sem2/PO_project/python/plants/nightshade.py
from __future__ import annotations

from python.organism import Plant


class NightShade(Plant):
    def __init__(self, x: int, y: int, game) -> None:
        super().__init__(99, 0, x, y, game)

    def get_color(self) -> tuple[int, int, int]:
        return (128, 0, 128)

    def get_organism(self):
        return self

    def spread(self, target_x: int, target_y: int):
        return NightShade(target_x, target_y, self.game)

    def collision(self, attacker) -> None:
        self.game.set_cell(attacker.get_x(), attacker.get_y(), None)
        if attacker in self.game.organisms:
            self.game.organisms.remove(attacker)
        super().collision(attacker)
